metadeploy/api/jobs.py

```diff
@@ -296,3 +262,66 @@ def expire_preflights():
 
 
 expire_preflights_job = job(expire_preflights)
+
+
+def create_scratch_org(*, plan_id, email, org_name, result_id):
+    plan = Plan.objects.get(id=plan_id)
+    org = ScratchOrgJob.objects.get(id=result_id)
+    repo_url = plan.version.product.repo_url
+    repo_owner, repo_name = extract_user_and_repo(repo_url)
+    commit_ish = plan.commit_ish
+    try:
+        with local_github_checkout(
+            repo_owner, repo_name, commit_ish=commit_ish
+        ) as repo_root:
+            scratch_org_config, _, org_config = create_scratch_org_on_sf(
+                repo_owner=repo_owner,
+                repo_name=repo_name,
+                repo_url=repo_url,
+                repo_branch=commit_ish,
+                email=email,
+                project_path=repo_root,
+                org_name=org_name,
+            )
+    except Exception as e:
+        org.fail(e)
+        return
+
+    org.complete(scratch_org_config.config)
+    fake_user = FakeUser(token=(org_config.access_token, org_config.refresh_token))
+
+    if plan.requires_preflight:
+        preflight_result = PreflightResult.objects.create(
+            user=None,
+            plan=plan,
+            organization_url=org_config.instance_url,
+            org_id=scratch_org_config.config["org_id"],
+        )
+        preflight(preflight_result.pk, forced_user=fake_user)
```

Comment:
If we're going to start this preflight automatically from the back-end, we'll need some way for the front end to find out about it (since they will not be subscribed to the preflight id WS channel). I think we could make that work by having the front-end find out via the `ORG_CHANGED` event and then subscribing if it's a preflight we don't know anything about yet, but I think a more robust solution would be to have a channel specifically for events related to a MD-created ScratchOrg?

Reply:
Notes from discussion: When this is triggered, @wlonk will send out a new `PREFLIGHT_STARTED` WS event on the _ScratchOrgJob_ channel, with the serialized Preflight. Then the front-end will subscribe to the preflight for further updates.